x-pack/filebeat/input/gcs/scheduler.go

```diff
@@ -141,9 +141,18 @@ func fetchJobID(workerId int, bucketName string, objectName string) string {
 }
 
 func (s *scheduler) createJobs(objects []*storage.ObjectAttrs, log *logp.Logger) []*job {
+	//nolint:prealloc // No need to preallocate the slice
 	var jobs []*job
-
+	fileSelectorLen := len(s.src.FileSelectors)
```

Comment:
Why is this done?

Reply:
Just so that unnecessary calls are not made to the method for the regex match if file selectors are not defined.

Reply:
`len` is not really a call.